src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+
+export interface ErrorBoundaryProps {
+  /**
+   *  Optional fallback component to render in case an error is thrown in the wrapped component
+   */
+  fallback?: React.ReactNode,
```

Comment:
This is a really clever way to handle a component failing to render. I've never thought about this approach